docs/js/features/odata/generic-http-client.md

```diff
@@ -0,0 +1,83 @@
+For each OData service, a separate client needs to be generated which defines the types and methods for this service.
+The client takes care of deserializing your request to a URL and JSON payload as well as serializing the response.
+The generated payload and URL are passed to the Generic HTTP Client.
+
+**Generic HTTP Client:** Adds SAP infrastructure specific functionality on top of a standard HTTP Client.
+All OData services use the same generic HTTP client so it contains no service specific information.
+The central aspect of this layer is connectivity based on [destination lookup](../connectivity/destination.md), [On-Premise](../connectivity/on-premise.md) connection via the connectivity service and [web proxy handling](../connectivity/proxy.md).
+In the end, all information from the destination, connectivity service and proxy configuration ends up in header-fields and [proxy-agents](https://www.npmjs.com/package/proxy-agent).
+The information goes one level down to the Axios client.
+
+**Axios HTTP Client:**  Is a widely used open source [HTTP client for node](https://www.npmjs.com/package/axios). 
+This client executes the actual HTTP requests in the end.
+ 
+## When To Use it
+
+In this section, we would like to provide some guidance when and when not to use the Generic HTTP client.
+You should consider the Generic HTTP client if:
+- ...you need to use an unsupported feature by the typed OData Client like `upsert` for example. 
+Then you get at least some convenience in the connectivity part.
+- ...you only want to ping a service or trigger a function import without a complicated payload. 
+In such a case the size of a full data model of the typed client is perhaps not worth the benefits. 
+- ...you want to call a non OData service which has no service definition.
+
+on the other hand you should consider the OData client if:
+- ...you have to build complicated filter, selection and/or expand conditions. 
+Here you will highly benefit from the help of the OData client.
+- ...you want to update or create new entities. 
+The OData Client has built-in `ETag` versions handling and takes also care of `CSRF` token fetching for you. 
+With the generic client you have to mange versions and tokens on your own.
+ 
+## How To Use it
+
+You call Generic HTTP client in the following way:
+
+```ts
+executeHttpRequest(destination,requestConfig)
+```
+
+where the `destination` argument is either a full destination object you have already fetched or an object containing a destination name and JWT.
+In the latter case the SDK fetches the destination as described [here](../connectivity/destination.md) for you.
+The request config argument contains the request configuration. 
+A minimal config would look like:
+```JSON
+{
+     method: HttpMethod.GET,
+     params: {
+          a: 'a',
+          b: 'b'
+     },
+    ...
+}
+```
+
+Note that you can also give values for `url` or `headers` in the request config. 
```

Comment:
I think url handling here is a crucial part because it is not 100% obvious. How does this relate to the destination? Can I only use relative URLs? What happens if I enter an absolute URL? (I don't know all these answers from the top of my head...)

A link to the docs of request config type interface would be helpful as well.

Reply:
added link and how the configs are merged.